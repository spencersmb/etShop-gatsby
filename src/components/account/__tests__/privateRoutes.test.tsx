import React from 'react'
import renderer from 'react-test-renderer'
import {
	render,
	cleanup
} from 'react-testing-library'
import 'jest-dom/extend-expect'
import PrivateRoute from '../privateRoutes'

afterEach(cleanup)
const testComp = () => (<div>Test Component</div>)

const hasUser = {
	Component: testComp,
	path: '/account/profile',
	location: {
		pathname: '/account/profile'
	},
	user: {
		email: 'spencer.bigum@gmail.com',
		firstName: 'spencer',
		lastName: 'spencer',
		token: '12345',
		gravatar: '',
		fbProfilePic: 'string'
	}
}

describe('Private Route Wrapper', () => {

	it('renders correctly', () => {
		const tree = renderer
			.create(
				<PrivateRoute {...hasUser}/>
			)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})

	it('Should render child component', () => {
		const modalRender = render(<PrivateRoute {...hasUser}/>)
		expect(modalRender.getByText('Test Component').innerHTML).toEqual('Test Component')
	})

})
