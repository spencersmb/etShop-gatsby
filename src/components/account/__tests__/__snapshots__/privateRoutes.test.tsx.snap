// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Private Route Wrapper renders correctly 1`] = `
<div>
  Test Component
</div>
`;
